tiny_dnn/optimizers/optimizer.h

```diff
@@ -135,6 +136,43 @@ struct adam : public stateful_optimizer<2> {
 };
 
 /**
+ * @brief [a new optimizer (2015)]
+ * @details [see Adam: A Method for Stochastic Optimization (Algorithm 2)
+ *               http://arxiv.org/abs/1412.6980]
+ *
+ */
+struct adamax : public stateful_optimizer<2> {
+  adamax()
+    : alpha(float_t(0.002)),
+      b1(float_t(0.9)),
+      b2(float_t(0.999)),
+      b1_t(b1),
+      eps(float_t(1e-8)) {}
+
+  void update(const vec_t &dW, vec_t &W, bool parallelize) {
+    vec_t &mt = get<0>(W);
+    vec_t &vt = get<1>(W);
+
+    for_i(parallelize, static_cast<int>(W.size()), [&](int i) {
```

Comment:
no need to cast here, just pass `size_t` to parallel for
`for_i(parallelize, W.size(), [&](size_t i)`